### PR TITLE
SAP HANA support 

### PR DESCRIPTION
**Pull request by VarshaRaviCV**

### CODE REVIEW

File: cvpysdk/backupsets/hanabackupset.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+
+class HANABackupset(Backupset):
+    """Derived class from Backupset Base class, representing a SAP HANA backupset,
+        and to perform operations on that backupset."""
+
+    def _get_backupset_properties(self):
+        """Derived class from Backupset Base class, representing a SAP HANA backupset,
+            and to perform operations on that backupset."""
+        super(HANABackupset, self)._get_backupset_properties()
```
Comment (GoelJatin) [inline cvpysdk/backupsets/hanabackupset.py]: No changes in the overridden method, please check, and remove this method

File: cvpysdk/backupsets/hanabackupset.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+                SDKException:
+                    if instance is not a string or object
+
+                    if response is empty
+
+                    if response is not success
+        """
+        from ..instance import Instance
+
+        if not (isinstance(instance, str) or isinstance(instance, basestring)):
```
Comment (GoelJatin) [inline cvpysdk/backupsets/hanabackupset.py]: Only use basestring

File: cvpysdk/backupsets/hanabackupset.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+HANABackupset:
+    _get_backupset_properties()     --  gets the properties of this subclient
+
+    restore()                       --  runs the restore job for specified backupset
+
+"""
+
+from __future__ import unicode_literals
+
```
Comment (GoelJatin) [inline cvpysdk/backupsets/hanabackupset.py]: basestring import missing

File: cvpysdk/instance.py [hunk 4]
```diff
@@ -235,7 +237,7 @@ def __init__(self, agent_object, instance_name, instance_id=None):
                     default: None
 
             Returns:
-                object - instance of the Instance class
+                object - instance of the Backupset class
```
Comment (GoelJatin) [inline cvpysdk/instance.py]: instance of the Instance class
please correct the doc-string

File: cvpysdk/instances/hanainstance.py [hunk 5]
```diff
@@ -0,0 +1,457 @@
...
+    _run_backup()                   --  runs full backup for this subclients and appends the
+                                            job object to the return list
+
+    backup()                        --  runs full backup for all subclients associated
+                                            with this instance
+
+    restore()                       --  runs the restore job for specified instance
+
+"""
+
```
Comment (GoelJatin) [inline cvpysdk/instances/hanainstance.py]: unicode_literals, and basestring import missing

File: cvpysdk/instances/hanainstance.py [hunk 6]
```diff
@@ -0,0 +1,457 @@
...
+            backup_prefix = ""
+
+        databases = []
+        if backupset_name != "default":
+            databases.append(backupset_name)
+
+        if point_in_time is None:
+            pit = {}
+        else:
+            if not isinstance(point_in_time, str):
```
Comment (GoelJatin) [inline cvpysdk/instances/hanainstance.py]: only do basestring comparison

File: cvpysdk/subclients/hanasubclient.py [hunk 7]
```diff
@@ -0,0 +1,274 @@
...
+            Args:
+                value   (str)   -- Log backup Storage policy name to be assigned to subclient
+
+            Raises:
+                SDKException:
+                    if failed to update log backup storage policy name
+
+                    if log backup storage policy name is not in string format
+        """
+        if isinstance(value, str):
```
Comment (GoelJatin) [inline cvpysdk/subclients/hanasubclient.py]: basestring

**Comment by VarshaRaviCV** [comment]
Jatin, I have made all the changes you mentioned. Let me know if anything else is needed.

**Pull request merged**